docs/proposals/dynamic-app-parameters.md

```diff
@@ -0,0 +1,181 @@
+
+## Access control
+
+Access control to the resource the value will be governed by the restrictions set for the Application in the corresponding AppProject. If the AppProject does not allow access to the referenced resource on the destination cluster, then defining a dynamic parameter referring to a forbidden resource will throw an error during manifest generation (see below).
+
+A new field in the AppProject should be considered. Existing `resourceAllowList` and `resourceDenyList` fields define both, read and write access to certain resources or resource classes. However, 
+
+## Value determination interval
+
+As noted above, the value for a dynamic parameter will be determined at manifest rendering time (i.e. at the time when the helm template command is run by Argo CD). This means that a change to a referenced resource will not trigger an immediate re-rendering of the source chart. The application needs to be either refreshed manually, or its freshness must expire.
+
+## Cluster access and watches
+
+The value for a dynamic parameter will be taken from the cluster cache, and there will be no additional calls to the remote cluster API. In the first incarnation of this feature, no additional watches will be established, nor will there be any new hooks for detecting modifications to the cluster cache.
+
+## Error handling
+
+Generally, failure to resolve a dynamic parameter should lead to an error in manifest generation, preventing a sync. The following error situations, additional to generic error situations, could occur when resolving dynamic parameters:
+
+### Access is forbidden by the AppProject
+
+If a referenced resource is not allowed to be accessed through the AppProject, manifest generation would terminate with an error stating that access to the resource is forbidden by configuration, without disclosing any other error details to the user. Details can be logged instead.
+
+### Access is forbidden by Kubernetes
+
+When access to the resource is forbidden by Kubernetes, the resource will not exist in the cluster cache. In this case, a generic “Resource not found” error should be returned, without disclosing any other error details to the user. Details can be logged instead.
+
+### Unknown kind or group
+
+When the user references a kind or group that does not exist on the destination cluster (possibly a typo in the configuration, etc), a detailed error message may be displayed to the user, since it is assumed that the user would have access to the resource.
+
+### Referenced resource does not exist
+
+When the referenced resource does not exist, and path is not set, it will set the parameter’s value to false. If a path is given, an error is returned to the user that the referenced resource does not exist.
+
+### Field in referenced resource does not exist
```

Comment:
Totally reasonable for a 1.0 of this feature, but might worth considering something analogous to https://github.com/argoproj/argo-cd/issues/7767 in the future.

Use case: The target kind/instance is created by another App

Reply:
Thanks for your suggestion @beekhof. Thinking about this, the behaviour could also be dependent on a `default` field being set:

* If `default` is set to anything (including the empty string), it will be used as the parameter's value when the target resource's field (or the resource itself) does not exist
* If `default` is omitted from the dynamic parameter's definition, an error will occur during render time if the target resource's field (or the resource itself) does not exist

I assume this will be easy enough to implement to be part of v1.0. WDYT?